custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/impl/Builder.java

```diff
@@ -182,23 +188,27 @@ public void build() throws IOException, InterruptedException {
         // Build Jenkinsfile Runner if needed
         JenkinsfileRunnerSettings jenkinsfileRunner = config.buildSettings.getJenkinsfileRunner();
         if (jenkinsfileRunner != null) {
-            if (!jenkinsfileRunner.getSource().isNeedsBuild()) {
-                throw new IOException("Jenkinsfile Runner always requires build");
-            }
             if (!config.war.artifactId.equals("jenkins-war")) {
                 throw new IOException("Jenkinsfile Runner packager can package only 'jenkins-war' so far");
             }
 
-            String jenkinsVersion = ComponentReference.resolveFrom(config.war, true, versionOverrides).getVersion();
-            buildIfNeeded(jenkinsfileRunner.getSource(), "jar",
-                    Arrays.asList(
-                            "-Djenkins.version=" + jenkinsVersion
-                            /*, "-Djenkins.testharness.version=2.38"*/));
             File outputDir = config.buildSettings.getOutputDir();
 
-            org.apache.commons.io.FileUtils.copyDirectory(
-                    new File(buildRoot, jenkinsfileRunner.getSource().artifactId + "/app/target/appassembler"),
-                    new File(outputDir, "jenkinsfileRunner"));
+            if (jenkinsfileRunner.getSource().isNeedsBuild()) {
+                String jenkinsVersion = ComponentReference.resolveFrom(config.war, true, versionOverrides).getVersion();
+                buildIfNeeded(jenkinsfileRunner.getSource(), "jar",
+                        Arrays.asList(
+                                "-Djenkins.version=" + jenkinsVersion
+                                /*, "-Djenkins.testharness.version=2.38"*/));
```

Comment:
Better remove the commented code.

Reply:
Yes, it's not needed anymore after https://issues.jenkins-ci.org/browse/JENKINS-54424

Reply:
Removed.